arch/riscv/Kconfig

```diff
@@ -46,6 +46,18 @@ config INCLUDE_RESET_VECTOR
 	  Include the reset vector stub, which initializes the stack and
 	  prepares for running C code.
 
+config RISCV_SOC_HAS_CUSTOM_IRQ_LOCK_OPS
```

Comment:
why not make it generic in naming, similar to many others ..HAS_CUSTOM_... kconfigs? Eventually yhis could be used by other architectures, i.e. we have
- ARCH_HAS_CUSTOM_BUSY_WAIT
- CPU_HAS_CUSTOM_FIXED_SOC_MPU_REGIONS
- ARCH_HAS_CUSTOM_SWAP_TO_MAIN
.....

Reply:
Just to be clear, you are suggesting to make these options available for all the architectures and possibly move this to `arch/Kconfig` ?